src/imitation/scripts/train_adversarial.py

```diff
@@ -36,102 +38,151 @@ def save(trainer, save_path):
+            from duplicate keyword arguments because algorithm-specific keyword
+            arguments are already set by the `gail_*` and `airl_*` parameters
+            to this function.
+        discrim_kwargs: Keyword arguments for the `DiscrimNet` constructor.
 
     Returns:
-      A dictionary with two keys. "imit_stats" gives the return value of
-        `rollout_stats()` on rollouts test-reward-wrapped
-        environment, using the final policy (remember that the ground-truth reward
-        can be recovered from the "monitor_return" key). "expert_stats" gives the
-        return value of `rollout_stats()` on the expert demonstrations loaded from
-        `rollout_path`.
+        A dictionary with two keys. "imit_stats" gives the return value of
+        `rollout_stats()` on rollouts test-reward-wrapped environment, using the final
+        policy (remember that the ground-truth reward can be recovered from the
+        "monitor_return" key). "expert_stats" gives the return value of
+        `rollout_stats()` on the expert demonstrations loaded from `rollout_path`.
     """
+    assert os.path.exists(rollout_path)
     total_timesteps = int(total_timesteps)
 
     tf.logging.info("Logging to %s", log_dir)
+    logger.configure(log_dir, ["tensorboard", "stdout"])
     os.makedirs(log_dir, exist_ok=True)
     sacred_util.build_sacred_symlink(log_dir, _run)
 
-    # Calculate stats for expert rollouts. Used for plot and return value.
     expert_trajs = types.load(rollout_path)
-
     if n_expert_demos is not None:
         assert len(expert_trajs) >= n_expert_demos
         expert_trajs = expert_trajs[:n_expert_demos]
-
-    expert_stats = rollout.rollout_stats(expert_trajs)
+    expert_transitions = rollout.flatten_trajectories(expert_trajs)
 
     with networks.make_session():
         if init_tensorboard:
-            sb_tensorboard_dir = osp.join(log_dir, "sb_tb")
-            kwargs = init_trainer_kwargs
-            kwargs["init_rl_kwargs"] = kwargs.get("init_rl_kwargs", {})
-            kwargs["init_rl_kwargs"]["tensorboard_log"] = sb_tensorboard_dir
+            tensorboard_log = osp.join(log_dir, "sb_tb")
+        else:
+            tensorboard_log = None
+
+        venv = util.make_vec_env(
+            env_name,
+            num_vec,
+            seed=_seed,
+            parallel=parallel,
+            log_dir=log_dir,
+            max_episode_steps=max_episode_steps,
+        )
+
+        # TODO(shwang): Let's get rid of init_rl later on?
+        # It's really just a stub function now.
+        gen_policy = util.init_rl(
+            venv, verbose=1, tensorboard_log=tensorboard_log, **init_rl_kwargs
+        )
 
-        trainer = init_trainer(
-            env_name, expert_trajs, seed=_seed, log_dir=log_dir, **init_trainer_kwargs
+        # Convert Sacred's ReadOnlyDict to dict so we can modify it.
+        discrim_kwargs = dict(discrim_kwargs)
+        auto_algorithm_kwargs = {}
```

Comment:
I'm worried that we now need to keep this code up to date as we change the arguments GAIL/AIRL/DiscrimNet's accept. Sacred does tend to encourage WET code unfortunately.

One alternative would be to introduce `algorithm_kwargs`, `discrim_kwargs` config entries with keys e.g. "global", "airl", "gail". The config in "global" would then be merged with "airl" or "gail" as appropriate. You could then dispense with most this case switching logic.

Reply:
Good idea with the top-level kwargs keys. Going ahead and implementing that.

Reply:
I'm using shared instead of global because global is reserved keyword